apps/src/templates/instructions/teacherFeedback/TeacherFeedback.jsx

```diff
@@ -201,10 +198,15 @@ export class TeacherFeedback extends Component {
   renderCommentAreaHeaderForTeacher() {
     const keepWorkingEnabled = experiments.isEnabled(experiments.KEEP_WORKING);
 
+    // We hide this feature for contained levels because contained levels are currently not
+    // editable by students so setting the review state to keepWorking doesn't make sense.
+    const hasEditableReviewState =
+      keepWorkingEnabled && !this.props.hasContainedLevels;
+
```

Comment:
Do you think disabling keep working for contained levels is more of a policy thing or more of a UI thing?  If it's more policy, then this logic should probably go on the back-end.  (I am ok with either, just would like to hear about your thought process.)

Reply:
@jamescodeorg I've added a method to `level.rb` `can_have_feedback_review_state?` where the logic lives for determining if the review state checkbox should show. It's passed to the frontend through pageConstants.. I'm not sure if there's a better place to pass this through?